fca/src/main/java/org/nmdp/ngs/fca/Interval.java

```diff
@@ -0,0 +1,554 @@
+/*
+
+    ngs-fca  Formal concept analysis for genomics.
+    Copyright (c) 2014-2015 National Marrow Donor Program (NMDP)
+
+    This library is free software; you can redistribute it and/or modify it
+    under the terms of the GNU Lesser General Public License as published
+    by the Free Software Foundation; either version 3 of the License, or (at
+    your option) any later version.
+
+    This library is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; with out even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this library;  if not, write to the Free Software Foundation,
+    Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA.
+
+    > http://www.gnu.org/licenses/lgpl.html
+
+*/
+package org.nmdp.ngs.fca;
+
+import com.google.common.annotations.Beta;
+
+import com.google.common.base.Objects;
+
+import com.google.common.collect.Range;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import com.google.common.collect.BoundType;
+import com.google.common.collect.ImmutableRangeSet;
+
+/**
+ * For spatial or temporal intervals defined by lower and upper comparable
+ * endpoints. This class wraps com.google.common.collect.Range and is endowed
+ * with a dimension to support programmatic frameworks for intervals in
+ * multidimensional space or time.
+ * @param <C> comparable endpoint type
+ */
+public class Interval<C extends Comparable> extends PartiallyOrdered<Interval<C>>
+                                            implements Comparable<Interval<C>> {
+    private int dimension;
```

Comment:
Yes, the term 'dimension' merely reflects a naive state. I think it can be generalized from a scalar (int) to a set, in which case Interval is a Concept, but my brain isn't quite there. In the meantime I can't think of better names. Suggestions welcome.